analyzers/tests/SonarAnalyzer.Test/Rules/MethodsShouldNotHaveIdenticalImplementationsTest.cs

```diff
@@ -22,36 +22,151 @@
 using CS = SonarAnalyzer.Rules.CSharp;
 using VB = SonarAnalyzer.Rules.VisualBasic;
 
-namespace SonarAnalyzer.Test.Rules
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class MethodsShouldNotHaveIdenticalImplementationsTest
 {
-    [TestClass]
-    public class MethodsShouldNotHaveIdenticalImplementationsTest
-    {
-        private readonly VerifierBuilder builderCS = new VerifierBuilder<CS.MethodsShouldNotHaveIdenticalImplementations>();
-        private readonly VerifierBuilder builderVB = new VerifierBuilder<VB.MethodsShouldNotHaveIdenticalImplementations>();
+    private readonly VerifierBuilder builderCS = new VerifierBuilder<CS.MethodsShouldNotHaveIdenticalImplementations>();
+    private readonly VerifierBuilder builderVB = new VerifierBuilder<VB.MethodsShouldNotHaveIdenticalImplementations>();
+
+    [TestMethod]
+    public void MethodsShouldNotHaveIdenticalImplementations() =>
+        builderCS.AddPaths("MethodsShouldNotHaveIdenticalImplementations.cs").WithOptions(ParseOptionsHelper.FromCSharp8).Verify();
+
+    [DataTestMethod]
+    [DataRow("", "")]
+    [DataRow("where T: struct", "where T: struct")]
+    [DataRow("where T: class", "where T: class")]
+    [DataRow("where T: unmanaged", "where T: unmanaged")]
+    [DataRow("where T: new()", "where T: new()")]
+    [DataRow("where T: IEquatable<T>, IComparable",  "where T: System.IComparable, IEquatable<T>")]
+    public void MethodsShouldNotHaveIdenticalImplementations_MethodTypeParameters_NonCompliant(string constraint1, string constraint2) =>
+        builderCS.AddSnippet($$"""
+            using System;
+            public static class TypeConstraints
+            {
+                public static bool Compare1<T>(T? value1, T value2) {{constraint1}} // Secondary
+                {
+                    Console.WriteLine(value1);
+                    Console.WriteLine(value2);
+                    return true;
```

Comment:
Maybe reduce all the method bodies too a single line.

Reply:
The methods need a certain size. Otherwise, the duplication check is not executed.